src/libcore/result.rs

```diff
@@ -820,6 +820,166 @@ impl<T, E> Result<T, E> {
     }
 }
 
+impl<T: Copy, E> Result<&T, E> {
+    /// Maps a `Result<&T, E>` to a `Result<T, E>` by copying the contents of the
+    /// `Ok` part.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(result_copied)]
+    /// let val = 12;
+    /// let x = Ok(&val);
+    /// assert_eq!(x, Ok(&12));
+    /// let copied = x.copied();
+    /// assert_eq!(copied, Ok(12));
+    /// ```
+    #[unstable(feature = "result_copied", reason = "newly added", issue = "63168")]
+    pub fn copied(self) -> Result<T, E> {
```

Comment:
My gut tells me that this should be called `copied_ok` so that there can be a `copied` that will copy both sides in one go.

Reply:
I'm not entirely sure about the usefulness. From my experience, I usually only need `copied_ok` and `cloned_ok`. But indeed having `copied`, `copied_ok`, and `copied_err` is more consistent. I've added them in the update.